src/crepe/facade/SDLContext.cpp

```diff
@@ -11,30 +11,28 @@
 #include <cstddef>
 #include <cstdint>
 #include <functional>
+#include <iostream>
 #include <memory>
 #include <stdexcept>
 
 #include "../api/Camera.h"
 #include "../api/Color.h"
 #include "../api/Config.h"
 #include "../api/Sprite.h"
-#include "../api/Texture.h"
 #include "../util/Log.h"
+#include "manager/Manager.h"
+#include "manager/Mediator.h"
 
 #include "SDLContext.h"
+#include "Texture.h"
 #include "types.h"
 
 using namespace crepe;
 using namespace std;
 
-SDLContext & SDLContext::get_instance() {
-	static SDLContext instance;
-	return instance;
-}
-
-SDLContext::SDLContext() {
+SDLContext::SDLContext(Mediator & mediator) : Manager(mediator) {
 	dbg_trace();
-
+	mediator.sdl_context = *this;
```

Comment:
Ik zou deze wel helemaal aan het einde van de constructor doen want je gooit mogelijk exceptions in je constructor